autoload/goyo.vim

```diff
@@ -33,8 +33,12 @@ function! s:get_color(group, attr)
 endfunction
 
 function! s:set_color(group, attr, color)
-  let gui = a:color =~ '^#'
-  execute printf('hi %s %s%s=%s', a:group, gui ? 'gui' : 'cterm', a:attr, a:color)
+  if empty(a:attr)
```

Comment:
I see your point. Then we should just check the var `termguicolors` to check if the user is using gui colors or not

```
function! s:set_color(group, attr, color)
    execute printf('hi %s %s%s=%s', a:group, &tgc ? 'gui' : 'cterm', a:attr, a:color)
endfunction
```